src/lib.rs

```diff
@@ -248,12 +248,36 @@ pub fn path_validation(project_path: &Path) -> Result<PathBuf> {
     use shellexpand::tilde;
 
     let expanded_path_str = tilde(project_path.to_string_lossy().as_ref()).to_string();
-    let project_path: PathBuf = expanded_path_str
+    let mut project_path: PathBuf = expanded_path_str
         .parse()
         .map_err(|_| Error::WrongExpandUser)?;
 
-    let result_path = project_path.canonicalize()?;
-    Ok(result_path)
+    if !project_path.exists() {
```

Comment:
Let's check if it's a directory, since the existence part is managed by `canonicalize`. If it is a file, let's return an error